sharedresource/v1alpha1/types_shared_configmap.go

```diff
@@ -0,0 +1,72 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SharedConfigMap allows a ConfigMap to be shared across namespaces.
+// Pods can mount the shared ConfigMap by adding a CSI volume to the pod specification using the
+// "csi.sharedresource.openshift.io" CSI driver and a reference to the SharedConfigMap in the volume attributes:
+//
+// spec:
+//  volumes:
+//  - name: shared-configmap
+//    csi:
+//      driver: csi.sharedresource.openshift.io
+//      volumeAttributes:
+//        sharedConfigMap: my-share
+//
+// For the mount to be successful, the pod's service account must be granted permission to 'use' the named SharedConfigMap object
+// within its namespace with an appropriate Role and RoleBinding. For compactness, here are example `oc` invocations for creating
+// such Role and RoleBinding objects.
+//
+//  `oc create role shared-resource-my-share --verb=use --resource=sharedconfigmaps.sharedresource.openshift.io --resource-name=my-share`
+//  `oc create rolebinding shared-resource-my-share --role=shared-resource-my-share --serviceaccount=my-namespace:default`
+//
+// Administrators can create separate Roles and RoleBindings for their users to be able the list and/or view the
```

Comment:
What do you see the default as?  Autthenticated list/watch of  SharedConfigMaps and SharedSecrets?

Reply:
I'll step in for this one (after discussing this with @coreydaley on slack today, as well has prior discussion I've had with @adambkaplan )

So yeah @coreydaley can augment the godoc here @deads2k, providing a compact summary of the following:
- we envisioned some aggregation to the default roles  .... for this one, I believe we are talking the `view` role
- with that, users can get (that is missing in this current version), list, and watch `SharedConfigMap` and `SharedSecret`
- so they can know which actual `ConfigMap` or `Secret` is being shared via a get on the `SharedConfigMap` or `SharedSecret`
- but they cannot of course see inside the `ConfigMap` or `Secret` unless they have sufficient permissions to those objects in the namespaces they reside
- minor sidebar -  `oc describe ... ` does a get under the covers

Any issues or things we missed there @deads2k ?

thanks

Reply:
In putting ^^ together @deads2k I am working off of https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles and hence believe we could aggregate get/list/watch perms for our 2 new *cluster scoped* CRDs here, assuming we all agree that is "safe", and I am in fact interpreting that doc correctly.

Reply:
@deads2k Updated with default permissions information. ptal